numba/datamodel/models.py

```diff
@@ -491,6 +490,17 @@ def from_data(self, builder, value):
         return NotImplemented
 
 
+@register_default(types.NestedArray)
+class NestedArrayModel(ArrayModel):
+    def __init__(self, dmm, fe_type):
+        self._be_type = dmm.lookup(fe_type.dtype).get_data_type()
+        super(NestedArrayModel, self).__init__(dmm, fe_type)
+
+    def get_data_type(self):
```

Comment:
I think that `sentry_record_type` is doing the right thing - after some further investigation, I've found that the `NestedArrayModel` needs some other methods defining to make them consistent with how the nested arrays are implemented.

I will add some more test cases and fix the `NestedArrayModel`.